templates/repo/diff/box.tmpl

```diff
@@ -195,6 +197,33 @@
 											{{ctx.Locale.Tr "repo.diff.bin_not_shown"}}
 										{{end}}
 									</div>
+								{{else if and $file.SubmoduleInfo}}
+									<div class="tw-p-3">{{svg "octicon-file-submodule"}} {{$submodule := $file.SubmoduleInfo -}}
+										{{- $refURL := $submodule.RefURL $.Repository.FullName -}}
+										{{- $submoduleName := $file.Name -}}
+										{{- $refShortID := ShortSha $submodule.RefID -}}
+										{{- if $file.IsDeleted -}}
+											{{- ctx.Locale.Tr "repo.diff.submodule_deleted" $submoduleName $refShortID -}}
+										{{- else if $file.IsCreated -}}
+											{{- $refAddedSubModule := $submoduleName -}}
+											{{- $refAddedAt := $refShortID -}}
+											{{- if $refURL -}}
+												{{- $refAddedSubModule = HTMLFormat `<a href="%s">%s</a>` $refURL $submoduleName -}}
+												{{- $refAddedAt = HTMLFormat `<a href="%s/commit/%s">%s</a>` $refURL (PathEscape $submodule.RefID) $refShortID -}}
+											{{- end -}}
+											{{- ctx.Locale.Tr "repo.diff.submodule_added" $refAddedSubModule $refAddedAt -}}
+										{{- else -}}
+											{{- $refUpdatedSubModule := $submoduleName -}}
+											{{- $refUpdatedFrom := ShortSha $submodule.PreviousRefID -}}
+											{{- $refUpdatedTo := $refShortID -}}
+											{{- if $refURL -}}
```

Comment:
This changes the logic I had in the original template, which was to construct a comparison URL for the two commits. Submodules are frequently used as a sort of ad-hoc package manager, so viewing the comparison URL helps get a sense of the changes in the diff.